# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLTypes.h
# Commit: e1b0906509ef
# Full Hash: e1b0906509efa6433978b9f980ec639b7eafbc6d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix non-webgl CI tests. r=handyman
#   
#   * Revert some partial webgl+oop+vr code.
#   * More missing FuncScope.
#   * Fix compile errors.
# ==============================================================================

diff -r 7e2a2b1b416f -r e1b0906509ef dom/canvas/WebGLTypes.h
--- a/dom/canvas/WebGLTypes.h	Wed Jan 08 22:19:23 2020 +0000
+++ b/dom/canvas/WebGLTypes.h	Wed Jan 08 22:19:26 2020 +0000
@@ -414,8 +414,8 @@
   static auto From(const U& val) {
     return From(val.x, val.y);
   }
-  template <>
-  static auto From(const gfx::IntSize& val) {
+  template <typename U>
+  static auto FromSize(const U& val) {
     return From(val.width, val.height);
   }
 
@@ -528,8 +528,8 @@
   uint32_t maxTexCubeSize = 0;
   uint32_t maxVertexAttribs = 0;
   std::array<uint32_t, 2> maxViewportDims = {};
-  std::array<float, 2> pointSizeRange = {1, 1};
-  std::array<float, 2> lineWidthRange = {1, 1};
+  std::array<float, 2> pointSizeRange = {{1, 1}};
+  std::array<float, 2> lineWidthRange = {{1, 1}};
 
   // WebGL 2
   uint32_t maxTexArrayLayers = 0;